### DISCUSSION TITLE
Multiple permissions for single route

### DISCUSSION THREAD

**Discussion by perfectime94** [maxcountryman/axum-login#138] [General]
Sorry if this isn't the best place to ask this question, Is possible to allow multiple permissions for a route? like 

`.route_layer(permission_required!(Backend, login_url = "/login", "restricted.read", "protected.read"))`

I looked at macro documentation and seems like it's possible (with limited rust knowledge). I tried with permissions example and on a custom bin that im making, i can provide it if needed

**Comment by maxcountryman** [comment]
The syntax in your example works, but it's important to point out that will require matching all permissions specified. If you want to match any of the permissions, you'd need to write your own middleware (tho that shouldn't be too difficult).

**Reply by perfectime94** [reply]
thanks for the clarification i will look at current middleware to see how to adapt it. Have a nice day

**Reply by maxcountryman** [reply]
If you don't want to write your middleware entirely from scratch, you can also use the lower-level [`predicated_required`](https://docs.rs/axum-login/latest/axum_login/macro.predicate_required.html) macro, which is what `permission_required` and `login_required` are based on.
